app/tag-management/annotation/annotation.component.ts

```diff
@@ -178,4 +270,35 @@ export class AnnotationComponent implements OnInit, OnDestroy {
     );
     return this.translatedResponse;
   }
+
+  private deleteTag(tag: AnnotationTag) {
+    let parentElement = tag.nativeElement.parentElement;
+    parentElement.innerHTML = parentElement.innerHTML.replace(tag.nativeElement.outerHTML, tag.nativeElement.innerHTML);
+    if(tag.isRelatedTo()) {
+      tag.undrawConnection(this.canvas);
+      tag.removeRelation();
+    }
+    let index = this.tagsInDocument.indexOf(tag);
+    this.tagsInDocument.splice(index, 1);
+    // reset the native Element of each tag, since it got removed by replacing the innerHTML
+    let tagsToReset: NodeListOf<HTMLElement> = parentElement.getElementsByTagName('span');
+    for(let i = 0; i < tagsToReset.length; i++) {
+      if('tagId' in tagsToReset[i].dataset) {
+        let tag = this.tagsInDocument.find((t: AnnotationTag) => t.id === +tagsToReset[i].dataset['tagId']);
+        tag.nativeElement = tagsToReset[i];
+      }
+    }
+  }
+
+  private static findNonWordCharacters(s: string) {
```

Comment:
__[tslint]__
* Declaration of private static member function not allowed to appear after declaration of private instance member function

<!-- Comment by [SideCI](https://sideci.com) -->